control-plane/config-entries/controllersv2/mesh_gateway_controller.go

```diff
@@ -126,7 +124,11 @@ func (r *MeshGatewayController) onCreateUpdate(ctx context.Context, req ctrl.Req
 		return fmt.Errorf("unable to create role binding: %w", err)
 	}
 
-	// TODO NET-6393
+	//Create Service
+	err = r.opIfNewOrOwned(ctx, resource, &corev1.Service{}, builder.Service(), upsertOp)
```

Comment:
`builderService()` isn't currently implemented. I'd expect this implementation to live in `gateways/service.go`, analogous to [gateways/serviceaccount.go](https://github.com/hashicorp/consul-k8s/blob/278509109e664014280bba062754c87c7d2e4f56/control-plane/gateways/serviceaccount.go)

Reply:
Whooops, it didn't 🤦‍♀️